SimG4Core/PrintGeomInfo/src/PrintGeomInfoAction.cc

```diff
@@ -173,7 +173,7 @@ void PrintGeomInfoAction::dumpG4LVLeaf(G4LogicalVolume *lv,
   //--- If a volume is placed n types as daughter of this LV, it should only be counted once
   std::map<G4LogicalVolume *, unsigned int> lvCount;
   std::map<G4LogicalVolume *, unsigned int>::const_iterator cite;
-  for (int ii = 0; ii < lv->GetNoDaughters(); ii++) {
+  for (size_t ii = 0; ii < lv->GetNoDaughters(); ii++) {
```

Comment:
as discussed elsewhere I prefer "int" to unsigned if operations (such as --) are involved.
So in my opinion this one should be modified using the pattern in the other below and those left unchanged or all modified as
```
for (int ii = 0, siz = lv->GetNoDaughters(); ii < siz;  ++ii)
```

Reply:
@VinInn , this code itself is rarely used  for tests only, so here we only remove compilation warnings. However, can you, please, clarify, what you mean:

  - it is more correct semantics?
  - potential problem with --i?
  - or potentially more performant code with int instead of size_t?

This may useful for other classes where we use such loops frequently.

Reply:
there was a thread with David at some point.
In general unsigned should not be used for arithmetics and not even for size: alo the C++ committee now agree that was a mistake to use unsigned for sizes.
Definetively -- can be a problem as it DOES NOT become negative
unsigned shall be used only for bit manipulation or when "wrapping" is intended (modular arithmetics)
Everybody agrees in particular that the index of a loop SHALL be a signed integer